libraries/ESP8266WiFi/src/include/WiFiState.h

```diff
@@ -0,0 +1,23 @@
+
+#ifndef WIFISTATE_H_
+#define WIFISTATE_H_
+
+#include <user_interface.h>
+#include <ESP8266WiFiType.h>
+
+struct WiFiState
+{
+    uint32_t crc;
+    struct
+    {
+        bool persistent;
+        WiFiMode_t mode;
+        uint8_t channel;
+        station_config fwconfig;
+        ip_info ip;
+        ip_addr_t dns[2];
+        ip_addr_t ntp[2];
+    } state;
```

Comment:
Ordering members from big to small will often lead to smaller struct sizes.
Also the members are not initialized in the constructor.
For example channel is only set if there was a successful connect and/or the active mode was WIFI_STA when shutdown and fwconfig is only set when in STA mode at shutdown.
But these are being used at wifi.begin.

I have not looked in much detail to see if the settings are somewhere read from flash, but I can imagine that would result in very hard to reproduce results.

Reply:
Members are initialized in shutdown only (now). The whole structure is valid only when crc is matching, and crc is calculated when all arguments are set.